crates/revm/src/executor.rs

```diff
@@ -322,7 +322,15 @@ where
                             success: false,
                             cumulative_gas_used,
                             logs: vec![],
-                            deposit_nonce: None,
+                            // Deposit nonces are only recorded after Regolith
+                            deposit_nonce: if self
+                                .chain_spec
+                                .is_fork_active_at_timestamp(Hardfork::Regolith, block.timestamp)
+                            {
+                                Some(old_sender_info.nonce)
+                            } else {
+                                None
+                            },
```

Comment:
[`.then(|| ...)`](https://doc.rust-lang.org/std/primitive.bool.html#method.then) style nit to make this more idiomatic
```suggestion
                            deposit_nonce: self
                                .chain_spec
                                .is_fork_active_at_timestamp(Hardfork::Regolith, block.timestamp)
                                .then(|| old_sender_info.nonce),
```